### ISSUE TITLE
VSCode hangs when expanding a folder in a workspace with multiple isfs connections configured

### ISSUE THREAD

**Issue by swk314159** [intersystems-community/vscode-objectscript#1429]
Since this past Friday (September 6, 2024), my usual VSCode workspace has been unable to open folders in any of my IRIS namespaces (ObjectScript files or web files).  Instead, when clicking on the folder, the folder shows the spinning progress wheel, but the progress wheel is never resolved, and the contents of the folder are never shown.

Note that the rest of VSCode functions normally - I can still edit files that live in folders on my local machine, for instance - but anything that needs to be accessed through an isfs connection stops working.

After some experimentation, I managed to narrow this down to the following minimal workspace configuration:
```
{
	"folders": [
		{
			"name": "local:XX6MEUI",
			"uri": "isfs://local2023:xx6meui/"
		},
		{
			"name": "local:XX6MEUI webfiles",
			"uri": "isfs://local2023:xx6meui/?csp"
		},
	],
	"settings": {
		"intersystems.servers": {
			"local2023": {
				"webServer": {
					"scheme": "http",
					"host": "localhost",
					"port": 52774
				},
				"username": "skanaski",
			},
		},
	}
}
```

If only one of those isfs connections is included in the workspace, the connection functions as normal: folders can be opened, files edited, etc.  If both are configured, neither works.  Also note that the issue is not specific to these two isfs connections.  Any two isfs connections causes this issue.

In case it ends up being relevant, here is the settings.json for my user:

```
{
    "workbench.colorTheme": "InterSystems Default Dark",
    "json.maxItemsComputed": 100000,
    "window.zoomLevel": -0.5227586988632231,
    "workbench.iconTheme": "vscode-icons",
    "objectscript.export": {
        "folder": "src",
        "addCategory": false,
        "map": {},
        "atelier": true,
        "generated": true,
        "filter": "",
        "exactFilter": "",
        "category": "*",
        "noStorage": false,
        "dontExportIfNoChanges": false,
        "maxConcurrentConnections": 0,
        "mapped": true
    },
    "files.trimTrailingWhitespace": true,
    "files.trimFinalNewlines": true,
    "vsicons.associations.files": [
        {
            "icon": "iris",
            "extensions": ["cls", "rtn", "mac","inc","csp","csr"],
            "format": "svg"
        }
    ],
    // Path to store the icon (in Perforce workspace) - uses "/" syntax without the actual folder for icons
    "vsicons.customIconFolderPath": "C:/VSCode/",
    "vsicons.dontShowNewVersionMessage": true,
    "typescript.updateImportsOnFileMove.enabled": "always",
}
```

See also https://intersystems.stackenterprise.co/questions/2568

**Comment by gjsjohnmurray** [comment]
Please see https://github.com/intersystems-community/vscode-objectscript/issues/1428#issuecomment-2359268381 as your issue may be related to that one, which showed up after VS Code 1.93 shipped widely.

**Comment by gjsjohnmurray** [comment]
We have just published v2.12.8 so please update and report back here whether or not your problem still occurs.

**Comment by swk314159** [comment]
Updating to v2.12.8 has resolved the issue, thank you!
